test/integration/consul-container/test/resource/resource_test.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package resource
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	libcluster "github.com/hashicorp/consul/test/integration/consul-container/libs/cluster"
+	libtopology "github.com/hashicorp/consul/test/integration/consul-container/libs/topology"
+)
+
+const (
+	RESOURCE_FILE_PATH_ON_HOST      = "../../../../../command/resource/testdata/demo.hcl"
+	RESOURCE_FILE_PATH_ON_CONTAINER = "/consul/data/demo.hcl"
+)
+
+func TestClientForwardToServer(t *testing.T) {
+	type operation struct {
+		action         func(*testing.T, libcluster.Agent, string) (int, string)
+		includeToken   bool
+		expectedCode   int
+		expectedMsg    string
+		expectedErrMsg string
+	}
+	type testCase struct {
+		description string
+		operations  []operation
+		aclEnabled  bool
+	}
+
+	testCases := []testCase{
+		{
+			description: "The read request should be routed to consul server agent",
+			operations: []operation{
+				{
+					action:         applyResource,
+					includeToken:   false,
+					expectedCode:   0,
+					expectedMsg:    "demo.v2.Artist 'korn' created.",
+					expectedErrMsg: "",
+				},
+			},
+			aclEnabled: false,
+		},
+		{
+			description: "The read request should be denied if missing token when ACL is enabled",
+			operations: []operation{
+				{
+					action:         applyResource,
+					includeToken:   false,
+					expectedCode:   1,
+					expectedMsg:    "",
+					expectedErrMsg: "failed getting authorizer: ACL not found",
+				},
+				{
+					action:         applyResource,
+					includeToken:   true,
+					expectedCode:   0,
+					expectedMsg:    "demo.v2.Artist 'korn' created.",
+					expectedErrMsg: "",
+				},
+			},
+			aclEnabled: true,
+		},
```

Comment:
Why create two operations since they're essentially two different test cases? I can understand if you're trying to optimize reusing the existing `setupClusterAndClient`, but then `t.Run(..)` it as a separate test case.

Reply:
This test has two steps, the first step is to test that if the request does not include the token, then it will be declined. The second request includes the token, so it's permitted.

Reply:
I'm not sure how this addresses the feedback. If the second step does not depend on the first step, then they're two separate test cases.

The description on L52 aligns with the operation on L55.  The operation on L62 does not align the the description on L52, hence is a separate test case. Maybe I'm missing something here?

Reply:
Ok, now I got your point, that makes sense, I'll separate them.